app/controllers/cinema_halls_controller.rb

```diff
@@ -4,40 +4,40 @@ class CinemaHallsController < ApplicationController
   include JSONAPI::Fetching
 
   def index
-    @cinema_halls = CinemaHall.all
-    render jsonapi: @cinema_halls
+    cinema_halls = CinemaHall.all
+    render json: CinemaHalls::Representers::Multiple.new(cinema_halls).call
   end
 
   def create
-    @cinema_hall = CinemaHall.new(cinema_hall_params)
+    cinema_hall = CinemaHalls::UseCases::Create.new(params: cinema_hall_params).call
 
-    if @cinema_hall.save
-      render jsonapi: @cinema_halls,
+    if cinema_hall.valid?
+      render json: cinema_halls,
              status: :created
     else
-      render jsonapi: @cinema_halls.errors,
+      render json: cinema_halls.errors,
              status: :unprocessable_entity
     end
   end
 
   def show
-    @cinema_hall = CinemaHall.find(params[:id])
-    render jsonapi: @cinema_hall
+    cinema_hall = CinemaHalls::Representers::Single.new(cinema_hall).call
```

Comment:
I'm pretty sure that you don't have access to `cinema_hall` variable which have to passed to representer.